addons/sourcemod/scripting/MapNotification.sp

```diff
@@ -7,65 +7,120 @@ ConVar g_cvWebhook;
+	Format(sMap, sizeof(sMap), "Map: **%s**", sMapName);
 
-    char sPlayerCount[64];
-    int iClients = GetClientCount(false);
-    Format(sPlayerCount, sizeof(sPlayerCount), "Players: **%d/%d**", iClients, MaxClients);
+	char sPlayerCount[64];
+	int iClients = GetClientCount(false);
+	Format(sPlayerCount, sizeof(sPlayerCount), "Players: **%d/%d**", iClients, MaxClients);
 
-    char sQuickJoin[255];
-    char sIP[32] = "";
-    char sPort[310] = "";
-    ConVar cIP = FindConVar("net_public_adr");
-    ConVar cHostPort = FindConVar("hostport");
+	char sQuickJoin[255];
+	char sIP[32] = "";
+	char sPort[310] = "";
+	ConVar cIP = FindConVar("net_public_adr");
+	ConVar cHostPort = FindConVar("hostport");
 
-    if (cIP != null && cHostPort != null)
-    {
-        cIP.GetString(sIP, sizeof(sIP));
-        cHostPort.GetString(sPort, sizeof(sPort));
-    }
-    Format(sQuickJoin, sizeof(sQuickJoin), "Quick join: **steam://connect/%s:%s**", sIP, sPort);
+	if (cIP != null && cHostPort != null)
+	{
+		cIP.GetString(sIP, sizeof(sIP));
+		cHostPort.GetString(sPort, sizeof(sPort));
+	}
+	
+	Format(sQuickJoin, sizeof(sQuickJoin), "Quick join: **steam://connect/%s:%s**", sIP, sPort);
 
-    char szWebhookURL[1000];
-    g_cvWebhook.GetString(szWebhookURL, sizeof szWebhookURL);
+	char szWebhookURL[1000];
+	g_cvWebhook.GetString(szWebhookURL, sizeof szWebhookURL);
+	if(!szWebhookURL[0])
+	{
+		LogError("[MapNotifications] No webhook found or specified.");
+		return;
+	}
+	
+	char sMessage[4096];
+	Format(sMessage, sizeof(sMessage), ">>> %s\n%s\n%s\n%s", sMap, sPlayerCount, sQuickJoin, sTime);
 
-    char sMessage[4096];
-    Format(sMessage, sizeof(sMessage), ">>> %s\n%s\n%s\n%s", sMap, sPlayerCount, sQuickJoin, sTime);
-
-    Webhook webhook = new Webhook(sMessage);
-    webhook.Execute(szWebhookURL, OnWebHookExecuted);
-    delete webhook;
+	Webhook webhook = new Webhook(sMessage);
+	
+	DataPack pack = new DataPack();
+	pack.WriteCell(view_as<int>(webhook));
+	pack.WriteString(szWebhookURL);
+	
+	webhook.Execute(szWebhookURL, OnWebHookExecuted, pack);
 }
 
 public void OnWebHookExecuted(HTTPResponse response, DataPack pack)
 {
-    if (response.Status != HTTPStatus_OK)
-    {
-        LogError("Failed to send mapnotification webhook");
-    }
+	static int retries;
+	
+	pack.Reset();
+	Webhook hook = view_as<Webhook>(pack.ReadCell());
+	
+	if (response.Status != HTTPStatus_OK)
+	{
+		if(retries < 3)
+			PrintToServer("[MapNotifcations] Failed to send the webhook. Resending it .. (%d/3)", retries);
+			
+		if(retries >= 3)
+		{
+			PrintToServer("[MapNotifcations] Failed to send the webhook. Aborting after %d retries.", retries);
+			LogError("[MapNotifcations] Failed to send the webhook after %d retries.", retries);
+			delete pack;
+			delete hook;
+			return;
+		}
+		
+		char webhookURL[PLATFORM_MAX_PATH];
+		pack.ReadString(webhookURL, sizeof(webhookURL));
+		
+		DataPack newPack;
+		CreateDataTimer(0.5, ExecuteWebhook_Timer, newPack);
```

Comment:
put the `CreateDataTimer` after the `newPack.WriteString(webhookURL);`

Reply:
No, data has to be written to the datapack after CreateDataTimer, because the CreateDataTimer function creates the datapack itself, so normally we cannot pass data to the datapack if it was never created.